pkg/cnb/project_descriptor.go

```diff
@@ -105,6 +119,17 @@ func processFiles(appDir string, d build) error {
 	})
 }
 
+// Because CNB Project Descriptor v0.2 has two ways for defining environment variables.
+// see https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md#iobuildpacksbuildenv-optional
+// This function calculates the final environment variables
+func getEnvVariables(d descriptorV2) []envVariable {
```

Comment:
Can we add this instead as a method on `descriptorV2` so that consumers of  descriptorV2 don't need to care about the heirachy of env and can just ask ` descriptor.Env()`?

Reply:
Done